tests/integration/tls_tests/test_tls.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+import time
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.tls_tests.helpers import (
+    METADATA,
+    check_tls,
+    time_file_created,
+    time_process_started,
+)
+
+TLS_CERTIFICATES_APP_NAME = "tls-certificates-operator"
+DATABASE_APP_NAME = "mongodb-k8s"
+TLS_TEST_DATA = "tests/integration/tls_tests/data"
+EXTERNAL_CERT_PATH = "/etc/mongodb/external-ca.crt"
+INTERNAL_CERT_PATH = "/etc/mongodb/internal-ca.crt"
+DB_SERVICE = "mongod.service"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy three units of MongoDB and one unit of TLS."""
+    async with ops_test.fast_forward():
+        my_charm = await ops_test.build_charm(".")
+        resources = {"mongodb-image": METADATA["resources"]["mongodb-image"]["upstream-source"]}
+        await ops_test.model.deploy(my_charm, num_units=3, resources=resources)
+        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=2000)
```

Comment:
You could combine this wait with line 35. You may also want to deploy and relate everything, then wait for idle to make sure your hooks are more robust.